service/frontend/nexus_http_handler.go

```diff
@@ -0,0 +1,162 @@
+// The MIT License
+//
+// Copyright (c) 2023 Temporal Technologies Inc.  All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package frontend
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"path"
+	"strings"
+
+	"github.com/gorilla/mux"
+	"github.com/nexus-rpc/sdk-go/nexus"
+	"go.temporal.io/server/api/matchingservice/v1"
+	"go.temporal.io/server/common/authorization"
+	"go.temporal.io/server/common/log"
+	"go.temporal.io/server/common/log/tag"
+	"go.temporal.io/server/common/metrics"
+	"go.temporal.io/server/common/namespace"
+	commonnexus "go.temporal.io/server/common/nexus"
+	"go.temporal.io/server/common/rpc"
+	"google.golang.org/grpc/credentials"
+)
+
+type NexusHTTPHandler struct {
+	logger       log.Logger
+	nexusHandler http.Handler
+	auth         *authorization.Interceptor
```

Comment:
Personally, I think it is a little confusing to reuse the gRPC interceptor here since we aren't using its `Intercept` functionality. Maybe it would make more sense to extract the `TLS` / `AuthInfo` / `Claims` handling logic to a separate component and inject that here and into the authorization interceptor?

Reply:
That's what I had before, I made an `authorization.Policy` struct but others didn't like it.
I think using the interceptor here is fine.

Reply:
Could you give a quick TL;DR of the objections to that approach? Just to satisfy my own curiosity.

Reply:
<img width="1693" alt="image" src="https://github.com/temporalio/temporal/assets/52304/addbcc00-1785-4963-a9b9-073bdc192c97">